contrib/mathjax-compat/mathjax-compat.js

```diff
@@ -0,0 +1,28 @@
+/* global katex */
+
+function replaceMathJaxInline(node) {
+    const katexinline = document.createElement("span");
+    katexinline.setAttribute("class", "inline-equation");
+    katexinline.innerHTML = katex.renderToString(node.text);
+    node.parentNode.replaceChild(katexinline, node);
+}
+
+function replaceMathJaxDisplay(node) {
+    const katexinline = document.createElement("div");
+    katexinline.setAttribute("class", "equation");
+    katexinline.innerHTML = "\\displaystyle " + katex.renderToString(node.text);
+    node.parentNode.replaceChild(katexinline, node);
+}
+
+{
+    const l = document.scripts.length;
+    for (let i = 0; i < l; i++) {
+        const s = document.scripts[i];
+        if (s.type === "math/tex") {
+            replaceMathJaxInline(s);
+        }
+        if (s.type === "math/tex; display") {
```

Comment:
You're right, there are definitely some bugs in this code. There at least ought to be a "mode=" before the display, and probably ought to compare case-insensitively. Unfortunately, I'm having trouble finding how MathJax does this. I'll keep looking.